### PR TITLE
Implement Alluxio POSIX API master stressbench test

### PR DESCRIPTION
**Pull request by LuQQiu**
### What changes are proposed in this pull request?

Add Alluxio POSIX API support for Master Stressbench and MaxThroughput tests.

### Why are the changes needed?

To have apple to apple comparison between Alluxio POSIX API / Native API / HDFS API interns of metadata read/write performance.

### Does this PR introduce any user facing changes?

Users can run MasterStressbench and MaxThroughput tests.

### CODE REVIEW

File: stress/shell/src/main/java/alluxio/stress/cli/StressMasterBench.java [hunk 1]
```diff
@@ -172,14 +179,13 @@ public void prepare() throws Exception {
...
+    LOG.info("Using {} to perform the test.", mParameters.mClientType);
     if (mParameters.mClientType == FileSystemClientType.ALLUXIO_HDFS) {
-      LOG.info("Using ALLUXIO HDFS Compatible API to perform the test.");
       mCachedFs = new FileSystem[mParameters.mClients];
       for (int i = 0; i < mCachedFs.length; i++) {
         mCachedFs[i] = FileSystem.get(new URI(mParameters.mBasePath), hdfsConf);
       }
-    } else {
-      LOG.info("Using ALLUXIO Native API to perform the test.");
+    } else if (mParameters.mClientType == FileSystemClientType.ALLUXIO_NATIVE) {
```
Comment (yyongycy) [inline stress/shell/src/main/java/alluxio/stress/cli/StressMasterBench.java:188]: What about if the clientType is other type than the two?

**Review by yyongycy — APPROVED** [review]
LGTM, left some minor questions

File: stress/shell/src/main/java/alluxio/stress/cli/StressMasterBench.java (hunk 1 above)
Comment (LuQQiu) [reply stress/shell/src/main/java/alluxio/stress/cli/StressMasterBench.java:188]: other client types e.g. Posix  API does not need the prepare stage

**Comment by LuQQiu** [comment]
Error: 2.982 [ERROR] Failures: 
Error: 2.982 [ERROR] alluxio.client.fs.LoadMetadataIntegrationTest.absentCacheMustCache
Error: 2.982 [ERROR]   Run 1: LoadMetadataIntegrationTest.absentCacheMustCache:205->checkListStatus:514->checkFunctionCall:538 Expected to be take between 500 and 750.0. actual duration (ms): 844
Error: 2.982 [ERROR]   Run 2: LoadMetadataIntegrationTest.absentCacheMustCache:205->checkListStatus:514->checkFunctionCall:538 Expected to be take between 500 and 750.0. actual duration (ms): 793

**Comment by LuQQiu** [comment]
Errors: 
Error: 1.039 [ERROR]   CheckConsistencyCommandIntegrationTest>AbstractFileSystemShellTest.beforeClass:122 » Runtime

**Review by dbw9580 — CHANGES_REQUESTED** [review]

File: stress/shell/src/main/java/alluxio/stress/cli/StressMasterBench.java [hunk 2]
```diff
@@ -702,4 +712,105 @@ protected void applyOperation(long counter) throws IOException, AlluxioException
...
+          break;
+        default:
+          throw new IllegalStateException("Unknown operation: " + mParameters.mOperation);
+      }
+    }
+
+    private java.nio.file.Path alluxioToFusePath(Path alluxioFullPath) {
+      String fuseMount = alluxio.conf.Configuration.getString(PropertyKey.FUSE_MOUNT_POINT);
+      String alluxioMountPath = alluxio.conf.Configuration
+          .getString(PropertyKey.FUSE_MOUNT_ALLUXIO_PATH);
```
[the remark this answers is not in the document]
Comment (LuQQiu) [reply stress/shell/src/main/java/alluxio/stress/cli/StressMasterBench.java:810]: yeah it's possible that user has different mount points, but currently alluxio configuration only supports one fuse mount `alluxio.fuse.mount.point=<only_one_fuse_mount_allowed`

I expect users use
```
bin/alluxio runClass alluxio.stress.cli.StressMasterBench --base alluxio:///stress-fuse .... --conf alluxio.fuse.mount.point=<MOUNT_POINT_A> --conf alluxio.fuse.mount.alluxio.path=<MOUNT_ALLUXIO_PATH_A>
```
to point to a specific fuse mount point

File: stress/shell/src/main/java/alluxio/stress/cli/StressMasterBench.java [hunk 3]
```diff
@@ -702,4 +712,105 @@ protected void applyOperation(long counter) throws IOException, AlluxioException
...
+    private AlluxioFuseBenchThread(BenchContext context) {
+      super(context);
+      mFuseFixedBasePath = alluxioToFusePath(mFixedBasePath);
+      mFuseBasePath = alluxioToFusePath(mBasePath);
+      try {
+        Files.createDirectories(mFuseFixedBasePath);
+        Files.createDirectories(mFuseBasePath);
+      } catch (IOException e) {
+        throw new RuntimeException(String.format("Failed to create directories %s %s",
+            mFuseFixedBasePath, mFuseBasePath));
```
[the remark this answers is not in the document]
Comment (LuQQiu) [reply stress/shell/src/main/java/alluxio/stress/cli/StressMasterBench.java:729]: good catch

File: stress/shell/src/main/java/alluxio/stress/cli/StressMasterBench.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (LuQQiu) [reply stress/shell/src/main/java/alluxio/stress/cli/StressMasterBench.java]: Use `basePath` to better address this problem.
And maxThroughput test can test against cloud disk storage to get performance understanding as well.

**Comment by LuQQiu** [comment]
@dbw9580 Thanks for reviewing the PR, address the comments, PTAL

**Review by dbw9580 — APPROVED** [review]
LGTM

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Pull request merged**